test/gtest/adaptiveavgpool.hpp

```diff
@@ -0,0 +1,377 @@
+    {
+        return os << " input_dims:" << tc.input_dims << " output_dims:" << tc.output_dims
+                  << "is_contiguous:" << tc.is_contiguous;
+    }
+
+    std::vector<size_t> GetInput() const { return input_dims; }
+    std::vector<size_t> GetOutput() const { return output_dims; }
+
+    std::vector<size_t> ComputeStrides(std::vector<size_t> inputDim) const
+    {
+        if(!is_contiguous)
+            std::swap(inputDim.front(), inputDim.back());
+        std::vector<size_t> strides(inputDim.size());
+        strides.back() = 1;
+        for(int i = inputDim.size() - 2; i >= 0; --i)
+            strides[i] = strides[i + 1] * inputDim[i + 1];
+        if(!is_contiguous)
+            std::swap(strides.front(), strides.back());
+        return strides;
+    }
+};
+
+inline std::vector<AdaptiveAvgPoolTestCase> AdaptiveAvgPoolTestConfigsFwdFp32()
+{
+    return {
+        {{64, 768, 17}, {10}, false},
+        {{64, 768, 17}, {10}, true},
+        {{64, 78, 17, 17}, {10, 10}, false},
+        {{64, 78, 17, 17}, {10, 10}, true},
+        {{6, 18, 18, 18, 18}, {5, 5, 5}, false},
+        {{6, 18, 18, 18, 18}, {5, 5, 5}, true},
+    };
+}
+
+inline std::vector<AdaptiveAvgPoolTestCase> AdaptiveAvgPoolTestConfigsFwdFp16()
+{
+    return {
+        {{64, 768, 17}, {10}, false},
+        {{64, 768, 17}, {10}, true},
+        {{64, 78, 17, 17}, {10, 10}, false},
+        {{64, 78, 17, 17}, {10, 10}, true},
+        {{6, 18, 18, 18, 18}, {5, 5, 5}, false},
+        {{6, 18, 18, 18, 18}, {5, 5, 5}, true},
+    };
+}
+
+inline std::vector<AdaptiveAvgPoolTestCase> AdaptiveAvgPoolTestConfigsFwdBfp16()
+{
+    return {
+        {{64, 768, 17}, {10}, false},
+        {{64, 768, 17}, {10}, true},
+        {{64, 78, 17, 17}, {10, 10}, false},
+        {{64, 78, 17, 17}, {10, 10}, true},
+        {{6, 18, 18, 18, 18}, {5, 5, 5}, false},
+        {{6, 18, 18, 18, 18}, {5, 5, 5}, true},
+    };
+}
+
+inline std::vector<AdaptiveAvgPoolTestCase> AdaptiveAvgPoolTestConfigsBwdFp32()
+{
+    return {
+        {{64, 768, 17}, {10}, false},
+        {{64, 768, 17}, {10}, true},
+        {{64, 206, 17, 17}, {10, 10}, false},
+        {{6, 18, 18, 18, 18}, {5, 5, 5}, false},
+        {{6, 18, 18, 18, 18}, {18, 18, 18}, true},
+    };
+}
+
+inline std::vector<AdaptiveAvgPoolTestCase> AdaptiveAvgPoolTestConfigsBwdFp16()
+{
+    return {
+        {{64, 768, 17}, {10}, false},
+        {{64, 768, 17}, {10}, true},
+        {{64, 28, 35, 35}, {35, 35}, false},
+        {{6, 28, 35, 35, 35}, {10, 10, 10}, false},
+        {{6, 28, 35, 35, 35}, {35, 35, 35}, true},
+    };
+}
+
+inline std::vector<AdaptiveAvgPoolTestCase> AdaptiveAvgPoolTestConfigsBwdBfp16()
+{
+    return {
+        {{64, 768, 17}, {10}, false},
+        {{64, 768, 17}, {10}, true},
+        {{64, 208, 9, 9}, {7, 7}, false},
+        {{6, 18, 12, 12, 12}, {5, 5, 5}, false},
+        {{6, 18, 12, 12, 12}, {12, 12, 12}, true},
+    };
+}
```

Comment:
Should this testcase genernerator be merged? They are the same.

Reply:
It should not. Each case has its own solver's pattern so I need to separate it to pass all

Reply:
Understood. They are not absolutely similar with each other.